batcher/test/endpoints/admin/test_add_remove_token_swap_pair.mligo

```diff
@@ -0,0 +1,241 @@
+#import "ligo-breathalyzer/lib/lib.mligo" "Breath"
+#import "./../../common/helpers.mligo" "Helpers"
+#import "./../../common/expect.mligo" "Expect"
+#import "../../../batcher.mligo" "Batcher"
+
+let get_tzbtc_eurl_swap_pair
+  (disabled: bool)
+  (oracle: address)
+  (storage: Batcher.Storage.t): (Batcher.valid_swap * Batcher.valid_swap_reduced) = 
+  let valid_tokens = storage.valid_tokens in
+  let tzbtc = Option.unopt (Map.find_opt "tzBTC" valid_tokens) in
+  let eurl = Option.unopt (Map.find_opt "EURL" valid_tokens) in
+  let swap : Batcher.swap = {
+      from = {
+        token = tzbtc;
+        amount = 1n;
+      };
+      to = eurl;
+  } in
+  let swap_reduced : Batcher.swap_reduced = {
+      to = eurl.name;
+      from = tzbtc.name;
+  } in
+  let valid_swap: Batcher.valid_swap = 
+  {
+    swap = swap;
+    oracle_address = oracle;
+    oracle_asset_name = "BTC-EURL";
+    oracle_precision = 6n;
+    is_disabled_for_deposits = disabled;
+  } in
+  let valid_swap_reduced =
+  {
+    swap = swap_reduced;
+    oracle_address = oracle;
+    oracle_asset_name = "BTC-EURL";
+    oracle_precision = 6n;
+    is_disabled_for_deposits = disabled;
+  } in
+  (valid_swap, valid_swap_reduced)
+
+let add_swap_pair_should_succeed_if_user_is_admin =
+  Breath.Model.case
+  "test add swap pair"
+  "should be successful if user is admin"
+    (fun (level: Breath.Logger.level) ->
+      let context = Helpers.test_context level in 
+      let batcher = context.contracts.batcher in
+      let bstorage = Breath.Contract.storage_of batcher in
+      let swap_pair_does_not_already_exist = Map.find_opt "tzBTC/EURL" bstorage.valid_swaps in
+      let (swap_pair, swap_pair_reduced) = get_tzbtc_eurl_swap_pair true context.contracts.oracle.originated_address bstorage in
+      let act_add_swap_pair = Breath.Context.act_as context.admin (fun (_u:unit) -> (Breath.Contract.transfer_to batcher (Add_token_swap_pair swap_pair) 0tez)) in
+      
+      let new_bstorage = Breath.Contract.storage_of batcher in
+      let added_swap_pair_reduced = Option.unopt (Map.find_opt "tzBTC/EURL" new_bstorage.valid_swaps) in
+
+      Breath.Result.reduce [
+        Breath.Assert.is_equal "swap pair should not already exist" None swap_pair_does_not_already_exist 
+        ; act_add_swap_pair
+        ; Breath.Assert.is_equal "swap pair should have been added" swap_pair_reduced added_swap_pair_reduced
+      ])
+
+let add_swap_pair_should_fail_if_user_is_non_admin =
+  Breath.Model.case
+  "test add swap pair"
+  "should be fail if user is non admin"
```

Comment:
```suggestion
  "should fail if user is non admin"
```

Reply:
I was surprised by this but you're the native speaker after all (and you use this form in several places as well). You do use "should fail" in lots of places though, so I'm not sure if it's a typo or not.

Reply:
Copy and paste laziness! Will fix